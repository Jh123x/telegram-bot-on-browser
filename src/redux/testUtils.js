import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { botSlice, defaultBotState } from "./botSlice";

export const generateDefaultState = () => {
  return {
    bot: defaultBotState,
  };
};

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: {
      bot: botSlice.reducer,
    },
    preloadedState,
  });
};

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: { bot: botSlice.reducer },
      preloadedState: preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
