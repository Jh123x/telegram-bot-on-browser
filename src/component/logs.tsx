import { TableContainer, TableRow, Table, TableHead, TableBody, TableCell } from '@mui/material';
import React from 'react';
import { useSelector } from 'react-redux';
import { BotWithConfig, Response } from '../redux/types';

export const LogBox = () => {
    const logs = useSelector<BotWithConfig, Response[]>((state) => state.bot.response)
    return <>
        <TableContainer style={{ maxHeight: "100%", overflow: 'scroll' }}>
            <Table>
                <TableHead>
                    <TableRow>
                        <TableCell key="date">Date</TableCell>
                        <TableCell key="username">User</TableCell>
                        <TableCell key="user-id">UserID</TableCell>
                        <TableCell key="message">Message</TableCell>
                    </TableRow>
                </TableHead>
                <TableBody>
                    {logs.map((response: Response) => {
                        const date = new Date(response.TimeStamp)
                        return <TableRow key={response.TimeStamp}>
                            <TableCell>{date.toISOString()}</TableCell>
                            <TableCell>{response.FromUser}</TableCell>
                            <TableCell>{`${response.UserID}`}</TableCell>
                            <TableCell>{response.Message}</TableCell>
                        </TableRow>
                    })}
                </TableBody>
            </Table>
        </TableContainer>
    </>
}