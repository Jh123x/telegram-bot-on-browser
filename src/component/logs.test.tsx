import { test, expect } from "@jest/globals";
import React from "react";
import { generateDefaultState, renderWithProviders, setupStore } from "../redux/testUtils.tsx";
import { LogBox } from "./logs.tsx";

test("renders log box interface correctly", () => {
    const store = setupStore(generateDefaultState());
    const component = renderWithProviders(<LogBox />, { store: store });
    expect(component).toMatchSnapshot();
});
