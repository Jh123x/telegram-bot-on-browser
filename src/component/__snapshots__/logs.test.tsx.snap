// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders log box interface correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiTableContainer-root css-70zvr5-MuiTableContainer-root"
        style="max-height: 100%; overflow: scroll;"
      >
        <table
          class="MuiTable-root css-1xwxv7r-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Date
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                User
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                UserID
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
                scope="col"
              >
                Message
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiTableContainer-root css-70zvr5-MuiTableContainer-root"
      style="max-height: 100%; overflow: scroll;"
    >
      <table
        class="MuiTable-root css-1xwxv7r-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
              scope="col"
            >
              Date
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
              scope="col"
            >
              User
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
              scope="col"
            >
              UserID
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
              scope="col"
            >
              Message
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
        />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
